### PR TITLE
Simplify the inbound context correlation API

### PR DESCRIPTION
**Pull request by sbuettner**

### CODE REVIEW

**Review by chillleader — APPROVED** [review]

**Review by markfarkas-camunda — APPROVED** [review]
LGTM!

**Review by igpetrov — APPROVED** [review]
LGTM, just a question to make sure we don't have a regression.
So, if we don't return a `CorrelationResult` now, can we still use `=correlation` in the return body expression.

**Comment by sbuettner** [comment]
@igpetrov I added a compatibility layer just now but as we upgraded to the latest FEEL engine an expression that uses it and tries to access a `null` value would not lead to any errors.

**Comment by igpetrov** [comment]
> @igpetrov I added a compatibility layer just now but as we upgraded to the latest FEEL engine an expression that uses it and tries to access a `null` value would not lead to any errors.

Sorry, I meant something different. There were numerous requests from customers that wanted to have `=correlation` result, that's why we initially introduced that. Just wanted to confirm whether we still support or want to support that.

**Review by chillleader — APPROVED** [review]

**Comment by sbuettner** [comment]
@igpetrov no, we dont want to support that for now.

File: parent/pom.xml [hunk 1]
```diff
@@ -44,6 +44,7 @@
     <version.java>17</version.java>
```
Comment (chillleader) [inline parent/pom.xml:44]: Shouldn't it be `java.version`?

Comment (chillleader) [reply parent/pom.xml:44]: Nevermind, it is consumed as `version.java` in camunda-release-parent

**Review by igpetrov — APPROVED** [review]
LGTM

**Pull request merged**